### PR TITLE
Fixed Maven proxy cache

### PR DESCRIPTION
**Pull request by g4s8**
#105 - fixed maven cache - save data to cache first, asked SO question https://stackoverflow.com/q/62405268/1723695

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#106](https://github.com/artipie/maven-adapter/issues/106) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/34cd33de-6571-41ea-8739-968135716214); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/maven/repository/StorageCacheTest.java [hunk 1]
```diff
@@ -29,55 +29,96 @@
...
             Matchers.equalTo(data)
         );
     }
 
     @Test
-    void loadsFromCache() throws Exception {
-        final Storage storage = new InMemoryStorage();
-        final Key key = new Key.From("cached");
+    @Disabled
+    void doesntSaveFailedPublisher(@TempDir final Path tmp) throws Exception {
```
Comment (Vatavuk) [inline src/test/java/com/artipie/maven/repository/StorageCacheTest.java:84]: @g4s8 add a puzzle for this disabled test

**Comment by g4s8** [comment]
@Vatavuk thanks, should be fixed in https://github.com/artipie/maven-adapter/pull/106/commits/600fdcfef295804a8040b12c5ad65ae0c07be0ed

**Comment by 0crat** [comment]
Job was finished in 22 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3  issues during next CR, thanks!

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**